### DISCUSSION TITLE
Failed to Connect

### DISCUSSION THREAD

**Discussion by NRutt2005** [siku2/hass-weatherlink#15] [Q&A]
I am getting a "failed to connect" when trying to trying to add the integration into the Home Assistant configuration. I've uninstalled and reinstalled the HACS repository and now I can't get the integration to add. Any thoughts?

**Comment by siku2** [comment]
The integration will give more details in the log in such cases.

**Reply by MacGeyer** [reply]
I get the same error, in the logs I can find following :

Cette erreur provient d'une intégration personnalisée

Logger: custom_components.weatherlink.config_flow
Source: helpers/aiohttp_client.py:73
Integration: WeatherLink (documentation, issues)
First occurred: 16:27:17 (5 occurrences)
Last logged: 16:33:29

failed to connect to 'http://10.0.0.100'
failed to connect to 'http://10.000.000.100'
Traceback (most recent call last):
  File "/config/custom_components/weatherlink/config_flow.py", line 45, in discover
    conditions = await session.current_conditions()
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/weatherlink/api/rest.py", line 73, in current_conditions
    return await self._request(CurrentConditions, EP_CURRENT_CONDITIONS)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/weatherlink/api/rest.py", line 69, in _request
    body = await resp.json()
           ^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/aiohttp_client.py", line 73, in json
    return await super().json(*args, loads=loads, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client_reqrep.py", line 1104, in json
    raise ContentTypeError(
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: text/html', url=URL('http://10.0.0.100/v1/current_conditions')

At http://10.0.0.100 :
![image](https://github.com/siku2/hass-weatherlink/assets/7631445/444c943e-f103-4c18-91d3-0d36fdfa8f48)

At http://10.0.0100/v1/current_conditions :
![image](https://github.com/siku2/hass-weatherlink/assets/7631445/57f039d6-40f8-4b5d-83f9-2e51c3a12522)

**Comment by MacGeyer** [comment]
I have updated my WeatherlinkIP ( https://support.davisinstruments.com/article/o3dfmlmgsp-weather-link-ip-data-logger-firmware-1-1-5-windows ) and restarted the device :
![image](https://github.com/siku2/hass-weatherlink/assets/7631445/361f7f56-5a61-43d3-bc88-8311c9ba7ca6)

This integration seams only be working with WeatherLink Live (WLL) not with WeatherLinkIP ( WLIP ) ( https://weatherlink.github.io/weatherlink-live-local-api/ ) !

**Reply by siku2** [reply]
I had no idea there even was such a thing. I will look into the possibility of supporting this as well.
